packages/scaffold-ui/src/views/w3m-select-addresses-view/index.ts

```diff
@@ -0,0 +1,150 @@
+  }
+
+  onSelectAll = (event: Event) => {
+    const checked = (event.target as HTMLInputElement).checked
+    this.selectAll = this.selectedAccounts.length === this.allAccounts.length
+    this.allAccounts.forEach(account => {
+      this.onSelect(account, checked)
+    })
+  }
+
+  onSelect = (account: AccountType, add: boolean) => {
+    if (add) {
+      this.selectedAccounts.push(account)
+    } else {
+      this.selectedAccounts = this.selectedAccounts.filter(a => a.address !== account.address)
+    }
+    if (this.selectedAccounts.length > 0) {
+      this.selectAll = this.selectedAccounts.length === this.allAccounts.length
+    }
+    this.requestUpdate()
+  }
+
+  // -- Render -------------------------------------------- //
+  public override render() {
+    return html`
+    <wui-flex justifyContent="center" .padding=${['xl', '0', 'xl', '0'] as const}>
+      <wui-banner-img imageSrc="${ifDefined(this.metadata?.icons[0])}" text="${ifDefined(
+        this.metadata?.url
+      )}" size="sm"></wui-banner>
+    </wui-flex>
+    <wui-flex .padding=${
+      ['0', 'xl', '0', 'xl'] as const
+    } flexDirection="row" justifyContent="space-between">
+        <wui-text variant="paragraph-400" color="fg-200">Select all</wui-text>
+        <input type="checkbox" .checked=${this.selectAll}  @click=${this.onSelectAll.bind(this)} />
+    </wui-flex>
+      <wui-flex flexDirection="column" .padding=${['l', 'xl', 'xl', 'xl'] as const}>
+        ${this.allAccounts.map(account => this.getAddressTemplate(account))}
+      </wui-flex>
+      <wui-flex .padding=${['l', 'xl', 'xl', 'xl'] as const} gap="s" justifyContent="space-between">
+        <wui-button
+          size="md"
+          ?fullwidth=${true}
+          variant="neutral"
+          @click=${this.onCancel.bind(this)}
+          data-testid="w3m-connecting-siwe-cancel"
+        >
+          Cancel
+        </wui-button>
+        <wui-button
+          size="md"
+          ?fullwidth=${true}
+          variant="main"
+          .disabled=${this.selectedAccounts.length === 0}
+          @click=${this.onContinue.bind(this)}
+          ?loading=${this.isApproving}
+        >
+          ${this.isApproving ? 'Signing...' : 'Continue'}
+        </wui-button>
+      </wui-flex>
+    `
+  }
+
+  // -- Private ------------------------------------------- //
+
+  getAddressTemplate = (account: AccountType) => {
+    const checked = this.selectedAccounts.some(_account => _account.address === account.address)
+
+    return html`<wui-list-account accountAddress="${account.address}" accountType="${account.type}">
+      <input
+        id="${account.address}"
+        slot="action"
+        type="checkbox"
+        .checked="${checked}"
+        @change="${this.handleClick(account)}"
+      />
+    </wui-list-account>`
+  }
+
+  private handleClick = (account: AccountType) => (event: Event) => {
+    const target = event.target as HTMLInputElement
+    this.onSelect?.({ ...account }, target?.checked)
+  }
```

Comment:
Let's avoid anonymous functions.
```suggestion
  private handleClick(account: AccountType) {
  return (event: Event) => {
    const target = event.target as HTMLInputElement;
    this.onSelect?.({ ...account }, target?.checked);
  };
}
```